src/schema/sources.ts

```diff
@@ -325,7 +342,10 @@ export const ensureSourcePermissions = async (
     const source = await ctx.con
       .getRepository(Source)
       .findOneByOrFail([{ id: sourceId }, { handle: sourceId }]);
-    if (await canAccessSource(ctx, source, permission)) {
+    const sourceMember = await ctx.con
```

Comment:
Can you unify it with the query above using `Promise.all` just to make it concurrent